src/Compilers/CSharp/Test/Symbol/Symbols/Source/NullablePublicAPITests.cs

```diff
@@ -727,5 +727,103 @@ public override void Initialize(AnalysisContext context)
+        }
+
+        [Fact]
+        [WorkItem(32661, "https://github.com/dotnet/roslyn/issues/32661")]
+        public void SuppressedTypedExpressionNullability()
+        {
+            var source = @"
+#pragma warning disable CS0219
+class C
+{
+    void M(string? s)
+    {
+        object o = s!;
+        string s1 = o!;
+        int? i = 1!;
+        D? d = this!;
+    }
+}
+class D
+{
+    public static implicit operator D?(C c) => throw null!;
+}
+";
+
+            var comp = CreateCompilation(source, options: WithNonNullTypesTrue(), parseOptions: TestOptions.Regular8WithNullableAnalysis);
+            comp.VerifyDiagnostics(
+                // (8,21): error CS0266: Cannot implicitly convert type 'object' to 'string'. An explicit conversion exists (are you missing a cast?)
+                //         string s1 = o!;
+                Diagnostic(ErrorCode.ERR_NoImplicitConvCast, "o").WithArguments("object", "string").WithLocation(8, 21));
+
+            var syntaxTree = comp.SyntaxTrees[0];
+            var root = syntaxTree.GetRoot();
+            var model = comp.GetSemanticModel(syntaxTree);
+            var notNull = new NullabilityInfo(PublicNullableAnnotation.NotAnnotated, PublicNullableFlowState.NotNull);
+            var @null = new NullabilityInfo(PublicNullableAnnotation.Annotated, PublicNullableFlowState.MaybeNull);
+            var notApplicable = new NullabilityInfo(PublicNullableAnnotation.NotApplicable, PublicNullableFlowState.NotApplicable);
+            var objectType = comp.GetSpecialType(SpecialType.System_Object);
+            var stringType = comp.GetSpecialType(SpecialType.System_String);
+            var intType = comp.GetSpecialType(SpecialType.System_Int32);
+            var nullableTType = comp.GetSpecialType(SpecialType.System_Nullable_T);
+            var nullableIntType = nullableTType.Construct(intType);
+            var cType = comp.GetTypeByMetadataName("C");
+            var dType = comp.GetTypeByMetadataName("D");
+
+            var suppressions = root.DescendantNodes().OfType<PostfixUnaryExpressionSyntax>().Where(p => p.IsKind(SyntaxKind.SuppressNullableWarningExpression)).ToArray();
+            AssertSuppressionNullability(model, suppressions[0], stringType, objectType, notNull, notNull); // o = s!
+            AssertSuppressionNullability(model, suppressions[1], objectType, stringType, notNull, notNull); // s1 = o!
+            AssertSuppressionNullability(model, suppressions[2], intType, nullableIntType, new NullabilityInfo(PublicNullableAnnotation.Annotated, PublicNullableFlowState.NotNull), notNull); // i = 1!
+            AssertSuppressionNullability(model, suppressions[3], cType, dType, notNull, notNull); // d = this!
+        }
+
+        private static void AssertSuppressionNullability(SemanticModel model, PostfixUnaryExpressionSyntax suppression, INamedTypeSymbol childType, INamedTypeSymbol type, NullabilityInfo nullability, NullabilityInfo childNullability)
```

Comment:
>INamedTypeSymbol childType, INamedTypeSymbol type, NullabilityInfo nullability, NullabilityInfo childNullability [](start = 120, length = 112)

The is a very confusing order for these parameters.  Would you consider combining them into pairs using tuples?